decoder.go

```diff
@@ -247,8 +253,15 @@ func (d *decoder) unmarshalMap(size uint, offset uint, result reflect.Value) (ui
 
 func (d *decoder) unmarshalPointer(size uint, offset uint, result reflect.Value) (uint, error) {
 	pointer, newOffset := d.decodePointer(size, offset)
-	_, err := d.decode(pointer, result)
-	return newOffset, err
+
+	switch result.Kind() {
+	default:
+		_, err := d.decode(pointer, result)
+		return newOffset, err
+	case reflect.Uintptr:
+		result.Set(reflect.ValueOf(uintptr(pointer)))
+		return newOffset, nil
+	}
```

Comment:
Is this change necessary? Wouldn't the same thing happen in the call to `.decode()` anyway?